src/Libraries/Nop.Core/Domain/Catalog/CatalogSettings.cs

```diff
@@ -539,5 +539,15 @@ public CatalogSettings()
         /// Gets or sets the attribute value display type when out of stock
         /// </summary>
         public AttributeValueOutOfStockDisplayType AttributeValueOutOfStockDisplayType { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether customer can search with manufacturer name
+        /// </summary>
+        public bool AllowCustomersToSearchWithManufacturerName { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether customer can search with category name
+        /// </summary>
+        public bool AllowCustomersToSearchWithCategoryName { get; set; }

```

Comment:
These settings must be filled during installation, let's set `true` by default. And don't forget about migration for these settings.